dxr/plugins/clang/dxr-index.cpp

```diff
@@ -1245,34 +1280,41 @@ class DXRIndexAction : public PluginASTAction {
       D.Report(DiagID) << output;
       return false;
     }
-    output = realpath(output.c_str(), NULL);
+    output = abs_output;
     output += "/";
+    FileInfo::output = output;
 
-    const char* tmp = getenv("DXR_CXX_CLANG_TEMP_FOLDER");
-    if(tmp){
+    // The temp directory for this plugin's output.
+    const char *tmp = getenv("DXR_CXX_CLANG_TEMP_FOLDER");
+    std::string tmpdir;
+    if (tmp)
       tmpdir = tmp;
-    }else
+    else
       tmpdir = output;
-
-    char* abs_tmpdir = realpath(tmpdir.c_str(), NULL);
-    if(!abs_tmpdir){
+    char *abs_tmpdir = realpath(tmpdir.c_str(), nullptr);
+    if (!abs_tmpdir) {
       DiagnosticsEngine &D = CI.getDiagnostics();
       unsigned DiagID = D.getCustomDiagID(DiagnosticsEngine::Error,
         "Temporary directory '%0' does not exist");
       D.Report(DiagID) << tmpdir;
       return false;
     }
-    tmpdir = realpath(tmpdir.c_str(), NULL);
+    tmpdir = abs_tmpdir;
     tmpdir += "/";
+    IndexConsumer::setTmpDir(tmpdir);
+
+    free(abs_src);
```

Comment:
We can't wait till here to free them all, because there are several early returns, right?